packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1465,22 +1528,27 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
       },
     );
 
+    if (widget._inputDecoration != null) {
+      result = InputDecorator(
+        decoration: widget._inputDecoration!,
+        isEmpty: widget._isEmpty,
+        isFocused: widget._isFocused,
+        child: result,
+      );
+    }
```

Comment:
I think we should move `DropdownButtonHideUnderline` to `DropdownButton` too, to maintains the tree order as before: `InputDecorator` -> `DropdownButtonHideUnderline` -> `DropdownButton`.

```suggestion
    if (widget._inputDecoration != null) {
      result = InputDecorator(
        decoration: widget._inputDecoration!,
        isEmpty: widget._isEmpty,
        isFocused: widget._isFocused,
        child: DropdownButtonHideUnderline(
          child: result,
        ),
      );
    }
```

Reply:
I don't wanna make too many changes that are not necessary to fix the issues unless it provides some benefit or the team says it could be required.

Reply:
> by "under", you meant to do InkWell(child: InputDecorator(...), ...)?

Yes, so InkWell size is equal to inputDecorator size which is the same in the native video in the description

Reply:
Oh right, obviously! Idk why misunderstood the code 😵‍💫